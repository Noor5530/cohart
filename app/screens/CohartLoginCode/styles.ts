import { appFonts } from 'components/text';
import { iconColor, primary, textColor } from 'config/colors';
import { StyleSheet } from 'react-native';
import { heightPercentageToDP as hp, widthPercentageToDP as wp } from 'react-native-responsive-screen';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'flex-end',
    flex: 1,
    alignItems: 'center',
    paddingBottom: hp(20),
  },
  checkContainer: {
    flexDirection: 'row',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: 23,
  },
  shortText: {
    width: 298.94,
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 18.0084,
    lineHeight: 22,
    alignItems: 'flex-end',
    textAlign: 'center',
    color: textColor,
    fontFamily: appFonts.InterRegular,
  },
  checkBox: {
    width: 20,
    height: 20,
    borderRadius: 5.7147,
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 0.476225,
    borderColor: iconColor,
    backgroundColor: textColor,
  },
  footer: {
    paddingLeft: 10,
    fontFamily: appFonts.InterRegular,
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 14,
    textAlign: 'center',
    lineHeight: 17,
    color: textColor,
  },
  termsConditionLine: {
    paddingLeft: 3,
    fontFamily: appFonts.InterRegular,
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 14,
    textAlign: 'center',
    lineHeight: 17,
    color: primary,
    textDecorationLine: 'underline',
    textDecorationColor: primary,
  },
  waitListButton: {
    fontFamily: appFonts.InterRegular,
    fontStyle: 'normal',
    fontWeight: '500',
    fontSize: 14,
    lineHeight: 17,
    alignItems: 'center',
    textAlign: 'center',
    color: primary,
    paddingTop: 10,
  },
  referral: {
    fontFamily: appFonts.InterRegular,
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 15.6501,
    lineHeight: 23,
    textAlign: 'center',
    color: primary,
    height: 23,
    marginTop: hp('11%'),
    textTransform: 'uppercase',
  },
  bottomText: {
    fontFamily: appFonts.InterRegular,
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 10.955,
    lineHeight: 16,
    textTransform: 'uppercase',
    position: 'absolute',
    bottom: 10,
    left: 41.24,
    color: textColor,
  },
  imageBackground: {
    height: hp(100),
    width: wp(100),
  },
  loginButton: {
    marginTop: hp('3'),
    height: 48,
    borderRadius: 30,
    paddingLeft: 15,
    paddingRight: 15,
    backgroundColor: primary,
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonTitle: {
    fontFamily: appFonts.InterRegular,
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: wp('3.8'),

    textAlign: 'center',
    color: iconColor,
  },
});

export default styles;
