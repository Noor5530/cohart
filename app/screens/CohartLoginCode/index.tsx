import { useRoute } from '@react-navigation/native';
import CodeTextInput from 'components/CodeTextInput';
import { RequestOTPType } from 'enums/request_otp_type';

import { IRequestOtpPayload } from 'models/actions/user';
import { LoginScreenRouteProps } from 'navigation/types';
import React from 'react';
import { ImageBackground, StatusBar, Text, TouchableOpacity, View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { useDispatch, useSelector } from 'react-redux';
import { resendOtp } from 'store/actions/userActions';

import { textColor } from '../../config/colors';
import AppState from '../../models/reducers';
import styles from './styles';

const CohartLoginCode: React.FC = () => {
    const route = useRoute<LoginScreenRouteProps>();
    const dispatch = useDispatch();

    const token = useSelector((state: AppState) => state.user.token);
    const resendOtpCodeWithPhoneNumber = () => {
        const payload : IRequestOtpPayload = {
            non_formatted_phone_number: '',
            country_code: route?.params?.data?.country_code,
            phone_number: route?.params?.data?.phone_number,
            registration_id: token,
            request_otp_type: RequestOTPType.PhoneNumber,
            resend_sms : true
        }

        dispatch(resendOtp(payload));
    };

    const resendOtpCodeWithEmail = () => {
        const payload: IRequestOtpPayload = {
          user_id: route?.params?.data.user_data?._id,
          request_otp_type: RequestOTPType.Email,
          country_code: "",
        };
        dispatch(resendOtp(payload));
    };


    return (
        <KeyboardAwareScrollView key="cohartLogin" style={{ flex: 1 }}>
            <StatusBar backgroundColor={textColor} barStyle="dark-content" />

            <ImageBackground
                style={styles.imageBackground}
                source={require('../../assets/LoginBackGround.png')}>
                <View style={styles.container}>
                    <Text style={styles.shortText}>
                        Enter the verification code sent to your phone{' '}
                    </Text>
                    <CodeTextInput
                        createWithCode={
                            route?.params?.data?.user_code ? true : false
                        }
                        phone_number={route?.params?.data?.phone_number}
                        countryCode={route?.params?.data?.country_code}
                        number={route?.params?.data?.number}
                    />

                    <View style={styles.checkContainer}>
                        <Text style={styles.footer}>
                            {'Didnâ€™t receive it?'}
                        </Text>
                    </View>
                    <TouchableOpacity
                        style={styles.loginButton}
                        onPress={resendOtpCodeWithPhoneNumber}>
                        <Text style={styles.buttonTitle}>
                            Resend via mobile number
                        </Text>
                    </TouchableOpacity>
                    {route?.params?.data?.user_data?.email &&
                    route?.params?.data?.user_data?.email != '' ? (
                        <TouchableOpacity
                            style={[
                                styles.loginButton,
                                { marginTop: hp('1.5') },
                            ]}
                            onPress={resendOtpCodeWithEmail}>

                            <Text style={styles.buttonTitle}>
                                Resend via registered email address
                            </Text>
                        </TouchableOpacity>
                    ) : null}

                    <Text style={styles.bottomText}>
                        Featured artist: alec demarco
                    </Text>
                </View>
            </ImageBackground>
        </KeyboardAwareScrollView>
    );
};

export default React.memo(CohartLoginCode);
