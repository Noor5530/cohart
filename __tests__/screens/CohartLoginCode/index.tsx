import 'react-native';
import React from 'react'; 
import CohartLoginCode from '../../../app/screens/CohartLoginCode';
import renderer from 'react-test-renderer';

jest.useFakeTimers()

test('renders correctly', () => {

    const tree = renderer.create(<CohartLoginCode />).toJSON();

    expect(tree).toMatchSnapshot();

});
