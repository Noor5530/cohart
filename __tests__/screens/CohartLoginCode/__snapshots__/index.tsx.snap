// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/LoginBackGround.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "paddingBottom": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#FFFFFF",
          "fontFamily": "Inter-Regular",
          "fontSize": 18.0084,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 22,
          "textAlign": "center",
          "width": 298.94,
        }
      }
    >
      Enter the verification code sent to your phone
       
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-between",
          "marginTop": undefined,
          "paddingHorizontal": undefined,
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        placeholder="X"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 14.4067,
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "height": 50,
            "margin": 0,
            "padding": 0,
            "paddingLeft": undefined,
            "width": undefined,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        placeholder="X"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 14.4067,
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "height": 50,
            "margin": 0,
            "padding": 0,
            "paddingLeft": undefined,
            "width": undefined,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        placeholder="X"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 14.4067,
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "height": 50,
            "margin": 0,
            "padding": 0,
            "paddingLeft": undefined,
            "width": undefined,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        placeholder="X"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 14.4067,
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "height": 50,
            "margin": 0,
            "padding": 0,
            "paddingLeft": undefined,
            "width": undefined,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        placeholder="X"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 14.4067,
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "height": 50,
            "margin": 0,
            "padding": 0,
            "paddingLeft": undefined,
            "width": undefined,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        keyboardType="number-pad"
        maxLength={1}
        onChangeText={[Function]}
        placeholder="X"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 14.4067,
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "height": 50,
            "margin": 0,
            "padding": 0,
            "paddingLeft": undefined,
            "width": undefined,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingTop": 23,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 17,
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        Didnâ€™t receive it?
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E6FF00",
          "borderRadius": 30,
          "height": 48,
          "justifyContent": "center",
          "marginTop": undefined,
          "opacity": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Inter-Regular",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Resend via mobile number
      </Text>
    </View>
    <Text
      style={
        Object {
          "bottom": 10,
          "color": "#FFFFFF",
          "fontFamily": "Inter-Regular",
          "fontSize": 10.955,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "left": 41.24,
          "lineHeight": 16,
          "position": "absolute",
          "textTransform": "uppercase",
        }
      }
    >
      Featured artist: alec demarco
    </Text>
  </View>
</View>
`;
